src/resolver/mod.rs

```diff
@@ -31,12 +33,12 @@ pub type Result<T> = ::std::result::Result<T, Error>;
 ///
 /// The `Resolver` side is a client of the `Executor`. Namerd work is performed on
 /// whatever thread the executor is spawned on.
-pub fn new(namerd: Namerd) -> (Resolver, Executor) {
+pub fn new<N: Namer + 'static + Send>(namer: N) -> (Resolver, Executor) {
```

Comment:
I'm very curious about the `'static` lifetime constraint here.  The intention of this function is that the Namer is moved into this function, boxed, and the box is moved into the Executor.  I am surprised that lifetimes come into play because, as far as I understand, Namer the just being moved around, not borrowed.

'static indicates that the Namer will live forever?  But since it is owned by the Executor, isn't it's lifetime the same as the Executor?